src/plugins/explore/public/components/results_summary/results_summary_panel.tsx

```diff
@@ -36,99 +28,60 @@ type AccordionState = 'closed' | 'open';
 
 const ACCORDION_STATE_LOCAL_STORAGE_KEY = 'resultsSummary.summaryAccordionState';
 
-export const convertResult = (body: IDataFrame) => {
-  const data = body as IDataFrame;
-  const hits: any[] = [];
-
-  if (data && data.fields && data.fields.length > 0) {
-    for (let index = 0; index < data.size; index++) {
-      const hit: { [key: string]: any } = {};
-      data.fields.forEach((field) => {
-        hit[field.name] = field.values[index];
-      });
-      hits.push({
-        _index: data.name,
-        _source: hit,
-      });
-    }
-  }
-
-  return hits;
-};
+const SUMMARY_REQUEST_SAMPLE_SIZE = 10;
 
 export const ResultsSummaryPanel: React.FC<ResultsSummaryPanelProps> = (props) => {
   const { services } = useOpenSearchDashboards<ExploreServices>();
-  const { core, http, isSummaryAgentAvailable$, data } = services;
-  const { query, search } = data;
+  const { core, http } = services;
 
-  const isSummaryAgentAvailable = useObservable(isSummaryAgentAvailable$ ?? of(false), false);
+  const isSummaryAgentAvailable = useSelector(selectSummaryAgentIsAvailable);
 
-  const [results, setResults] = useState<any[]>([]);
   const [accordionState, setAccordionState] = useState<AccordionState>('open');
   const [summary, setSummary] = useState(''); // store fetched data
   const [loading, setLoading] = useState(false); // track loading state
   const [feedback, setFeedback] = useState(FeedbackStatus.NONE);
-  const [isEnabledBySetting, setIsEnabledBySetting] = useState(false);
-  const selectedDataset = useRef(query.queryString.getQuery()?.dataset);
-  const queryState = useSelector((state: any) => state.query);
-  const status = useSelector((state: RootState) => state.ui.status);
-  const updateQueryState = useCallback((x: any) => {}, []); // FIXME
 
-  const isQueryDirty =
-    queryState.query && queryState.query !== data.query.queryString.getQuery().query;
+  const { reportMetric, reportCountMetric } = useMetrics(props.usageCollection);
+
+  const {
+    queryState,
+    status,
+    queryPromptState,
+    dataSetState,
+    queryResults,
+    editorMode,
+  } = useSelector((state: RootState) => ({
+    queryState: state.query,
+    status: state.queryEditor.queryStatus.status || QueryExecutionStatus.UNINITIALIZED,
+    queryPromptState: state.queryEditor.lastExecutedPrompt,
+    dataSetState: state.query.dataset,
+    queryResults: state.results[state.query.query as string]?.hits?.hits,
+    editorMode: state.queryEditor.editorMode,
+  }));
+
+  const assistantEnabled = useMemo(() => !!core.application.capabilities?.assistant?.enabled, [
+    core.application.capabilities?.assistant?.enabled,
+  ]);
```

Comment:
useMemo is useless here, we can simply do this:
```suggestion
  const assistantEnabled = !!core.application.capabilities?.assistant?.enabled;
```